ccd-adapter/src/main/java/uk/gov/hmcts/cmc/ccd/mapper/claimantresponse/ClaimantResponseMapper.java

```diff
@@ -29,11 +31,11 @@ public ClaimantResponseMapper(PaymentIntentionMapper paymentIntentionMapper) {
     }
 
     public CCDClaimantResponse to(Claim claim) {
+        requireNonNull(claim,"claim must not be null");
         final ClaimantResponse claimantResponse = claim.getClaimantResponse().orElse(null);
         if (null == claimantResponse) {
```

Comment:
Unwrapping an `Optional` to `null` and checking `null` looks code smell. Please either use `Optional.isPresent()` or `Optional.ifPresent()` etc.

Reply:
agreed.

Reply:
Although I agreed to this earlier, I think the existing code is valid. If I use , ifPresent() then I will have to move the entire logic in an inline block which I do not prefer. and I will still have to return null when ifPresent() is false.Alternatively if I just initialse claimantResponse in the ifPresent() then the code is still pretty much the same plus some extra lines. If I use isPresent() then I have to do all of the above + I have to fetch the claim.getClaimantResponse().get() .. which is an extra line of code. I feel the way it is written right now is fine.so I prefer to keep it this way.  Not sure what you mean by code smell. In the future try to explain things in technical terms rather than giving vague and metaphoric comments.